helixbass/tree-sitter-lint
# Basic fixing, in-memory entry points for rule-testing

In this PR:
- allow rules to declare themselves "fixable" and to specify fixes when reporting violations (which will be applied if running with `--fix`)
- add "in-memory" (vs touching the filesystem) entry points that take a slice of the file contents to lint
- use those in-memory entry points for the rule-testing helpers
- move the existing rule tests to be "unit tests"

To test:
If you run with `--fix` against a directory with files that contain `Default::default()` it should replace those with `_d()` and not report those as violations
If you do that without `--fix` it should report those as violations and not edit the files

Based on `initial-publish`
